gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/TopologyControllerTest.java

```diff
@@ -0,0 +1,105 @@
+import io.camunda.zeebe.gateway.protocol.rest.Partition.HealthEnum;
+import io.camunda.zeebe.gateway.protocol.rest.Partition.RoleEnum;
+import io.camunda.zeebe.gateway.protocol.rest.TopologyResponse;
+import io.camunda.zeebe.gateway.rest.util.TestApplication;
+import io.camunda.zeebe.protocol.record.PartitionHealthStatus;
+import io.camunda.zeebe.util.VersionUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+@SpringBootTest(
+    classes = {TestApplication.class, TopologyController.class},
+    webEnvironment = WebEnvironment.RANDOM_PORT)
+public class TopologyControllerTest {
+
+  @MockBean BrokerClient brokerClient;
+  @MockBean BrokerTopologyManager topologyManager;
+
+  @Autowired private WebTestClient webClient;
+
+  @BeforeEach
+  void setUp() {
+    Mockito.when(brokerClient.getTopologyManager()).thenReturn(topologyManager);
+  }
+
+  @Test
+  public void shouldGetTopology() {
+    // given
+    final String gatewayVersion = VersionUtil.getVersion();
+    final TopologyResponse topologyResponse = getTopologyResponse(gatewayVersion);
+    final BrokerClusterState brokerClusterState = getBrokerClusterState(gatewayVersion);
+    Mockito.when(brokerClient.getTopologyManager().getTopology()).thenReturn(brokerClusterState);
+
+    // when / then
+    webClient
+        .get()
+        .uri("api/v1/topology")
+        .header(HttpHeaders.ACCEPT, "application/json")
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectBody(TopologyResponse.class)
+        .isEqualTo(topologyResponse);
+  }
+
+  private TopologyResponse getTopologyResponse(final String gatewayVersion) {
+    final Partition partition = new Partition();
+    partition.setPartitionId(1);
+    partition.setRole(RoleEnum.LEADER);
+    partition.setHealth(HealthEnum.HEALTHY);
+
+    final BrokerInfo brokerInfo = new BrokerInfo();
+    brokerInfo.setNodeId(0);
+    brokerInfo.setHost("localhost");
+    brokerInfo.setPort(26501);
+    brokerInfo.addPartitionsItem(partition);
+    brokerInfo.setVersion(gatewayVersion);
+
+    final TopologyResponse topologyResponse = new TopologyResponse();
+    topologyResponse.setGatewayVersion(gatewayVersion);
+    topologyResponse.setClusterSize(1);
+    topologyResponse.setPartitionsCount(1);
+    topologyResponse.setReplicationFactor(3);
+    topologyResponse.addBrokersItem(brokerInfo);
+
+    return topologyResponse;
+  }
+
+  private BrokerClusterState getBrokerClusterState(final String gatewayVersion) {
+    final BrokerClusterStateImpl brokerClusterState = new BrokerClusterStateImpl();
```

Comment:
:x: I'm not a fan of accessing implementation classes directly from outside the module, since you're breaking the module's own abstractions. Instead, it might be better to create stubs out of the interfaces, or just mock them. The interfaces are the API, not the implementation classes. We don't want to couple these here, I think, even if it's for testing.